src/storage/src/statistics.rs

```diff
@@ -378,6 +457,44 @@ impl SinkStatisticsRecord {
         self.bytes_staged = 0;
         self.bytes_committed = 0;
     }
+
+    /// Reset counters so that we continue to ship diffs to the controller.
+    fn reset_counters(&mut self) {
+        self.messages_staged = 0;
+        self.messages_committed = 0;
+        self.bytes_staged = 0;
+        self.bytes_committed = 0;
+    }
+
+    /// Aggregate the other record into this one. This overrides gauges, but sums counter diffs.
+    fn ingest(&mut self, mut other: Self) {
```

Comment:
should this be named `incorporate` like the trait?

Reply:
its slightly different, because its summing counter diffs, but maybe? they is probably a better name

Reply:
actually you are right!